libs/components/datetime/testing/src/date-range-picker/date-range-picker-harness.ts

```diff
@@ -0,0 +1,194 @@
+import { HarnessPredicate } from '@angular/cdk/testing';
+import { SkyComponentHarness } from '@skyux/core/testing';
+// eslint-disable-next-line @nx/enforce-module-boundaries
+import { SkyDateRangeCalculatorId } from '@skyux/datetime';
+import { SkyInputBoxHarness } from '@skyux/forms/testing';
+
+import { SkyDatepickerHarness } from '../public-api';
+
+import { SkyDateRangePickerFilters } from './date-range-picker-harness.filters';
+
+/**
+ * Harness for interacting with date range picker components in tests.
+ */
+export class SkyDateRangePickerHarness extends SkyComponentHarness {
+  /**
+   * @internal
+   */
+  public static hostSelector = 'sky-date-range-picker';
+
+  #getStartDateInput = this.locatorFor(
+    SkyInputBoxHarness.with({
+      ancestor: '.sky-date-range-picker-start-date',
+    }),
+  );
+  #getEndDateInput = this.locatorFor(
+    SkyInputBoxHarness.with({
+      ancestor: '.sky-date-range-picker-end-date',
+    }),
+  );
+  #getCalculatorSelectInput = this.locatorFor(
+    SkyInputBoxHarness.with({
+      ancestor: '.sky-date-range-picker-select-calculator',
+    }),
+  );
+  #getCalculatorSelect = this.locatorFor(
```

Comment:
This is only used once in the harness, so we can probably put it inside the `selectCalculator` method, and use its input box harness to query the `select` element:
```
const select = await (
  await this.#getCalculatorIdInputBoxHarness()
).querySelector('select');
```